app/assets/javascripts/controllers/container_deployment/deploy-provider-controller.js

```diff
@@ -0,0 +1,247 @@
+        else if ($scope.data.provisionOn == 'noProvider') {
+          name = item.vmName;
+          var publicName = item.publicName;
+        }
+        nodes.push({
+          name: name,
+          id: id,
+          public_name: publicName,
+          roles: {
+            master: item.master,
+            node: item.node,
+            storage: item.storage,
+            master_lb: item.loadBalancer,
+            dns: item.dns,
+            etcd: item.etcd,
+            infrastructure: item.infrastructure
+          }
+        });
+      });
+      nodes = nodes.filter(function (node) {
+        return node.roles.master || node.roles.node || node.roles.storage
+      });
+      return nodes;
+    };
+
+    var create_deployment_resource = function() {
+      var method_types = {
+        existingVms: "existing_managed",
+        newVms: "provision",
+        noProvider: "unmanaged"
+      };
+
+      var resource = {
+        provider_name: $scope.data.providerName,
+        provider_type: $scope.data.providerType,
+        method_type: method_types[$scope.data.provisionOn],
+        rhsm_authentication: {
+          userid: $scope.data.rhnUsername,
+          password: $scope.data.rhnPassword,
+          rhsm_sku: $scope.data.rhnSKU,
+          rhsm_server: $scope.data.rhnSatelliteUrl
+        },
+        ssh_authentication: {
+          userid: $scope.data.deploymentUsername,
+          auth_key: $scope.data.deploymentKey
+        },
+        nodes: create_nodes_object(),
+        identity_authentication: create_auth_object()
+      };
+
+      if ($scope.data.provisionOn == 'existingVms') {
+        resource.underline_provider_id = $scope.data.existingProviderId;
+      } else if ($scope.data.provisionOn == 'newVms') {
+        resource.underline_provider_id = $scope.data.newVmProviderId;
+        resource.masters_creation_template_id = $scope.data.masterCreationTemplateId;
+        resource.nodes_creation_template_id = $scope.data.nodeCreationTemplateId;
+        resource.master_base_name = $scope.data.createMasterBaseName;
+        resource.node_base_name = $scope.data.createNodesBaseName;
+      }
+      return resource;
+    };
+
+    $scope.ready = false;
+
+    $scope.data = {};
+    $scope.deployComplete = false;
+    $scope.deployInProgress = false;
+
+    var startDeploy = function () {
+      $scope.deployInProgress = true;
+      $scope.deployComplete = false;
+      $scope.deploySuccess = false;
+      $scope.deployFailed = false;
+
+      var url = '/api/container_deployments';
+      var resource = create_deployment_resource();
+      API.post(url, {"action" : "create", "resource" : resource}).then(function (response) {
+        'use strict';
+        $scope.deployInProgress = false;
+        $scope.deployComplete = true;
+        $scope.deployFailed = response.error !== undefined;
+        if (response.error) {
+          if (response.error.message) {
+            $scope.deployFailureMessage = __(response.error.message);
```

Comment:
Sure, makes sense :) Unless there's like 2 possible error messages in which case a `switch` may make sense..